src/launcher/java/org/truffleruby/launcher/Launcher.java

```diff
@@ -173,8 +164,13 @@ public static void processArguments(CommandLineOptions config, String[] argument
             CommandLineParser.processEnvironmentVariable("TRUFFLERUBYOPT", config, false);
         }
 
-        if (!config.doesHaveScriptToRun() && !config.shouldUsePathScript() && System.console() != null) {
-            config.setUsePathScript("irb");
+        if (config.getOption(OptionsCatalog.EXECUTION_ACTION) == ExecutionAction.UNSET) {
+            if (System.console() != null) {
+                config.setOption(OptionsCatalog.EXECUTION_ACTION, ExecutionAction.PATH);
+                config.setOption(OptionsCatalog.TO_EXECUTE, "irb");
+            } else {
+                config.setOption(OptionsCatalog.EXECUTION_ACTION, ExecutionAction.STDIN);
+            }
```

Comment:
I expect to move this around a bit to make it more sharable with the GVM launcher. I already call it from both.